lib/compress.js

```diff
@@ -261,6 +262,24 @@ merge(Compressor.prototype, {
                         d.fixed = false;
                     }
                 }
+                if (node instanceof AST_Defun) {
+                    var d = node.name.definition();
+                    if (!toplevel && d.global || is_safe(d)) {
+                        d.fixed = false;
+                    } else {
+                        // Make a shallow copy of definition body but as
+                        // expression instead - ensure to clone whatever
+                        // child nodes you extract from this
```

Comment:
- `evaluate` currently cannot operate on `AST_Function` and will be an noop
- IIFE transform ensures removal of the original `AST_Defun` so no multiple reference issues
- only possible reuse is within `OPT(AST_Call)`, which I've just patched in https://github.com/mishoo/UglifyJS2/pull/1560/commits/5cb71547fb538d1dcd58967ca95b3097316d6a17